engine/src/main/java/org/terasology/rendering/world/WorldRenderer.java

```diff
@@ -33,6 +34,13 @@
  * the update and render methods as they are central to a rendering implementation.
  */
 public interface WorldRenderer {
+    // This is exposed so that controller-related functions can create callbacks.
+    // Ideally, these classes would be able to do so with dependency injection, but that
+    // is broken for at least FirstPersonHeldMountPointComponent.java. Could be reltaed to:
+    // https://github.com/MovingBlocks/Terasology/issues/2336A
+    // TODO: take this out and do dependency injection properly for any classes that depend on this being exposed.
+    public OpenVRProvider VR_PROVIDER = new OpenVRProvider();
```

Comment:
It's worse than that; I can get neither. The reference to `WorldRenderer` is null in this case. Here a bit of my Java ignorance is shining through, because,

* I'm sure that my public static instance to `VR_PROVIDER` is the same instance (otherwise the callback that I registered would not be hit), but
* I'm not 100% what that object's lifecycle is.

The best I can tell, 'WorldRendererImpl' is instantiated and VRProvider is instantiated for the first time when `WorldRendererImpl` accesses the instance, or when the program is first run; I'm not real sure on that part. Then that instance is accessible anywhere via `WorldRenderer.VR_PROVIDER`. Then, as long as `WorldRenderer.VR_PROVIDER` is reachable by some thread, that instance escapes garbage collection. I guess the exact same effect can be achieved with something like:

```
import org.terasology.rendering.world.WorldRenderer.
...
VR_PROVIDER; VR_PROVIDER.addControllerListener();
```

The code ended up how it did just because I was trying to get an `@In` reference to work and that's when I gave up. Anyway, that's a long way of saying that the code as it is written is very confusing and possibly a future minefield, and it would be good if we could find an alternative. Any ideas, anyone, on:

* how I can get dependency injection to work, or
* if there is an alternative?